### PR TITLE
DB-10690 fix duplicate FirstWriteToken

### PR DESCRIPTION
**Pull request by martinrupp**

### CODE REVIEW

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1952/

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1954/

File: splice_si_api/src/main/java/com/splicemachine/si/impl/server/SITransactor.java [hunk 1]
```diff
@@ -325,12 +327,12 @@ private void releaseLocksForKvBatch(Pair<KVPair, Lock>[] locks){
             conflictingChildren[i] = conflictResults.getChildConflicts();
             boolean addFirstOccurrenceToken = false;
 
-            if (!skipConflictDetection) {
+            if (checkedConflicts) {
```
Comment (arnaud-splice) [inline splice_si_api/src/main/java/com/splicemachine/si/impl/server/SITransactor.java:330]: Can you add a comment there explaining the defensive approach, as well a reference to a jira that tracks our efforts relaxing that defensive approach in the interest of performance?
Also, make sure that PhysicalDeletionIt still pass after this change.

**Review by arnaud-splice — APPROVED** [review]

**Review by msirek — APPROVED** [review]

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1960/

File: splice_si_api/src/main/java/com/splicemachine/si/impl/server/SITransactor.java (hunk 1 above)
Comment (martinrupp) [reply splice_si_api/src/main/java/com/splicemachine/si/impl/server/SITransactor.java:330]: done

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1962/

**Review by carolp-503 — APPROVED** [review]
+1

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1966/

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1969/

**Pull request merged**